lazy-pages/src/common.rs

```diff
@@ -219,25 +244,29 @@ impl LazyPagesExecutionContext {
 /// First part is always the same, so we can copy it to buffer
 /// once and then use it for all pages.
 #[derive(Debug)]
-pub(crate) struct PagePrefix {
-    buffer: Vec<u8>,
+pub struct PagePrefix {
+    storage_prefix: Vec<u8>,
 }
 
 impl PagePrefix {
     /// New page prefix from program prefix
-    pub fn new_from_program_prefix(mut program_prefix: Vec<u8>) -> Self {
-        program_prefix.extend_from_slice(&u32::MAX.to_le_bytes());
```

Comment:
this was made for performance in order to avoid redundant allocations, which you returns now

Reply:
I doubt it affects performance that much or does it at all

Reply:
May be) but then you should make all investigations and proof this. Because currently you changed working, more effective and already existing implementation to theoretically less effective. What the reason to do this?